xcp_d/utils/hcp2fmriprep.py

```diff
@@ -15,154 +15,226 @@
+    )
+    copy_dictionary[t1w_orig].append(template_to_t1w_fmriprep)
+
+    rh_midthickness_fmriprep = os.path.join(
+        anat_dir_fmriprep, f"{sub_id}_hemi-R_midthickness.surf.gii"
+    )
+    copy_dictionary[rh_midthickness_orig] = [rh_midthickness_fmriprep]
+    lh_midthickness_fmriprep = os.path.join(
+        anat_dir_fmriprep, f"{sub_id}_hemi-L_midthickness.surf.gii"
+    )
+    copy_dictionary[lh_midthickness_orig] = [lh_midthickness_fmriprep]
+    rh_inflated_fmriprep = os.path.join(anat_dir_fmriprep, f"{sub_id}_hemi-R_inflated.surf.gii")
+    copy_dictionary[rh_inflated_orig] = [rh_inflated_fmriprep]
+    lh_inflated_fmriprep = os.path.join(anat_dir_fmriprep, f"{sub_id}_hemi-L_inflated.surf.gii")
+    copy_dictionary[lh_inflated_orig] = [lh_inflated_fmriprep]
+
+    rh_pial_fmriprep = os.path.join(anat_dir_fmriprep, f"{sub_id}_hemi-R_pial.surf.gii")
+    copy_dictionary[rh_pial_orig] = [rh_pial_fmriprep]
+    lh_pial_fmriprep = os.path.join(anat_dir_fmriprep, f"{sub_id}_hemi-L_pial.surf.gii")
+    copy_dictionary[lh_pial_orig] = [lh_pial_fmriprep]
+
+    rh_wm_fmriprep = os.path.join(anat_dir_fmriprep, f"{sub_id}_hemi-R_smoothwm.surf.gii")
+    copy_dictionary[rh_wm_orig] = [rh_wm_fmriprep]
+    lh_wm_fmriprep = os.path.join(anat_dir_fmriprep, f"{sub_id}_hemi-L_smoothwm.surf.gii")
+    copy_dictionary[lh_wm_orig] = [lh_wm_fmriprep]
 
-    for i, j in zip(hcpfiles, newanatfiles):
-        copyfileobj_example(i, j)
     print("finished converting anat files")
-    # get the task files
 
-    taskx = glob.glob(hcp_dir + "/Results/*")
-    tasklistx = []
-    print(hcp_dir)
-    for j in taskx:
-        if j.endswith("RL") or j.endswith("LR"):
-            tasklistx.append(j)
+    # get the task files
+    subject_task_folders = sorted(glob.glob(os.path.join(in_dir, "Results", "*")))
+    subject_task_folders = [task for task in subject_task_folders if task.endswith(["RL", "LR"])]
 
     csf_mask = pkgrf("xcp_d", "/data/masks/csf.nii.gz")
     wm_mask = pkgrf("xcp_d", "/data/masks/wm.nii.gz")
-    for k in tasklistx:
-        idx = os.path.basename(k).split("_")
-        filenamex = os.path.basename(k)
+    for subject_task_folder in subject_task_folders:
+        _, task_name, acq_label = os.path.basename(subject_task_folder).split("_")
+        filenamex = os.path.basename(subject_task_folder)
+
+        # Find original task files
+        bold_nifti_orig = os.path.join(subject_task_folder, f"{filenamex}.nii.gz")
+        brainmask_orig_temp = os.path.join(subject_task_folder, "brainmask_fs.2.nii.gz")
+        boldref_orig = os.path.join(subject_task_folder, "SBRef_dc.nii.gz")
+        bold_cifti_orig = os.path.join(
+            subject_task_folder, f"{filenamex}_Atlas_MSMAll.dtseries.nii"
+        )
+
+        # Construct fMRIPrep task filenames
+        boldref_fmriprep = os.path.join(
+            func_dir_fmriprep,
+            f"{sub_id}_task-{task_name}_acq-{acq_label}_space-MNI152NLin6Asym_boldref.nii.gz",
```

Comment:
What is the acq label? Is it the phase-encoding direction? If so, we should change `acq` to `dir`.

Reply:
I've switched to dir. Can always switch back if it's a mistake.